### ISSUE TITLE
Issue in Travi

### ISSUE THREAD

**Issue by semirotta** [kolton/d2bot-with-kolbot#1055]
Almost all the time Travincal does this to my soso:
https://i.imgur.com/sJ0HqbO.png

I dont understand why the bot wont go and get merc, heal up do the precasts and go back in. It just stays there trying to "Orb" the enemy even though its Cold and Fire immune, which is what my merc would handle. But instead as my merc always seem to die in there because the bot doesnt know how to dodge the strikes as it used to for some reason and now it gets constantly stuck in that one immune. Every ducking morning when I come back its either tons of death messages in list or its still stuck in that one point wasting all my potions.

Any idea how to prevent this? I'd like to keep clearing travincal but at this moment, I think it just isnt worth it unless I am here following the bots run.

**Comment by DarkHorseDre** [comment]
please paste a pastebin link to your config.

I assume you have MercWatch AND telestomp enabled?
bot should return to town when it does due to merc watch and the inability to telestomp.

**Comment by semirotta** [comment]
I have both turned True and I have seen it go take the merc if it dies everywhere else but in there, it just stays and tries to orb it without the merc.
Here is hastelink: https://hastebin.com/yuqexonuru.php
I have turned Travincal off in this now because I took it off just few hours back. But only thing thats different in this is the Travincal from True to False, thats it.

**Comment by DarkHorseDre** [comment]
Is this the only cold and fire immune you come across?  would explain why this is the only problem you have..

looking at what you have set, you probably intend for orb to be used for all unless they are immune, then static them alternated with fireball.  in theory you can do this, but you've chosen some wonky skills:

youve got frozen armour as preattack skill - remove that - it will be cast by precast.js and recast when running out. 

youve got static set for immune skill - remove that - static is ineffective at killing as it is capped.
you have Config.CastStatic set - use that if you need static help.  I am not sure if this is causing an issue. it will either be cast inbetween using fireball or it is causing a problem or the bot is smart enough to not cast it once life has dropped to cap.

also i wouldn't set orb as an untimed skill as it does have a cooldown (1 or 2 secs i think). also the notes say that attackskill[4] should only be set if attackskill[3] is untimed, which it isnt. i found this caused problems when i tried last ladder.  

remove static and pre attack and set fireball as immune skill and see how it behaves. you really need a 3rd skill in there tho.

heres one i used before, check the notes (re: attackskill [5] cant be same as [2] - this caused me problems) I made also - there are other ways to do this, but:
    Config.AttackSkill[0] = -1; // Preattack skill.
    Config.AttackSkill[1] = 64; // Primary skill to bosses. 64 = orb
    Config.AttackSkill[2] = 45; // Primary untimed skill to bosses. Keep at -1 if Config.AttackSkill[1] is untimed skill. == ice
    Config.AttackSkill[3] = 64; // Primary skill to others. 64 = orb
    Config.AttackSkill[4] = 47; // Primary untimed skill to others. Keep at -1 if Config.AttackSkill[3] is untimed skill. == fireball
    Config.AttackSkill[5] = 47; // Secondary skill if monster is immune to primary.   // cant be same as [2]
    Config.AttackSkill[6] = 45; // Secondary untimed skill if monster is immune to primary untimed. == ice blast

check `d2bs/kolbot/libs/config/Templates/Attacks.txt` for examples.
heres one - change nova for fireball, and take out glacial and uv got something similar i think) - note that there is only one timed OR untimed skill set in this case, and how it flips from orb to nova to spike:
*** Dual element - frozen orb + nova (with glacial spike to lightning immune)
	Config.AttackSkill[0] = -1; // Preattack skill. Not implemented yet.
	Config.AttackSkill[1] = 64; // Primary skill to bosses.
	Config.AttackSkill[2] = 48; // Primary untimed skill to bosses. Keep at -1 if Config.AttackSkill[1] is untimed skill.
	Config.AttackSkill[3] = 64; // Primary skill to others.
	Config.AttackSkill[4] = 48; // Primary untimed skill to bosses. Keep at -1 if Config.AttackSkill[3] is untimed skill.
	Config.AttackSkill[5] = -1; // Secondary skill if monster is immune to primary.
	Config.AttackSkill[6] = 55; // Secondary untimed skill if monster is immune to primary untimed.

**Comment by semirotta** [comment]
Well the travi creature has been the only thing ive seen myself to be bugging like that.
All the configs are done by googling as how to, there isnt much guides around and to someone like me who has never used these things, the configs are confusing at times.
Also the untimed and timed, I have no idea how that works :'D I have not had any issues with this though. Other than the act4-5 crashings almost constantly and that travi bug. Otherwise it works fine.
This is what I have now:
	Config.AttackSkill[0] = -1; // Preattack skill.
	Config.AttackSkill[1] = 64; // Primary skill to bosses.
	Config.AttackSkill[2] = 64; // Primary untimed skill to bosses. Keep at -1 if Config.AttackSkill[1] is untimed skill.
	Config.AttackSkill[3] = 64; // Primary skill to others.
	Config.AttackSkill[4] = 64; // Primary untimed skill to others. Keep at -1 if Config.AttackSkill[3] is untimed skill.
	Config.AttackSkill[5] = -1; // Secondary skill if monster is immune to primary.
	Config.AttackSkill[6] = 47; // Secondary untimed skill if monster is immune to primary untimed.

	// Low mana skills - these will be used if main skills can't be cast.
	Config.LowManaSkill[0] = 0; // Timed low mana skill.
	Config.LowManaSkill[1] = 0; // Untimed low mana skill.

**Comment by 5noop** [comment]
You should set  `Config.LowManaSkill` to -1, because 0 is the normal attack. 
Orb is not an untimed skill so you should just set `Config.AttackSkill[2]` and `Config.AttackSkill[4]` to -1.
Finally as a general advice set a higher value for your chicken, 5 is way too low as the bot will not have time to leave before dying. 20+ would be better.

**Comment by semirotta** [comment]
Ah ok well luckily I rarely have situation where it chickens out and usually it doesnt even die. But I did the changes you guided me to do and we'll see if anything changes. I need to try keep looking at the bot when its in act3 travi to see if it changed anything at all. Might have just been an issue with the attacks who knows. Much appreciated (y)

**Comment by DarkHorseDre** [comment]
Good catch with lowmanaskill @5noop :)

**Comment by semirotta** [comment]
I have not had the travi issue again yet, got to keep watching ^^

**Comment by DarkHorseDre** [comment]
wait you said the problem went and then it came back? you must have changed something?

confirm you changed your skills as directed and post your config before going forward with this..

**Comment by semirotta** [comment]
No, I never said it went away. I just said I need to keep my eye out to follow if it happens again and it seems to still happen just like it used to. 
The skills are listed below and are the way I was told to change them

> 	Config.AttackSkill[0] = -1; // Preattack skill.
	Config.AttackSkill[1] = 64; // Primary skill to bosses.
	Config.AttackSkill[2] = -1; // Primary untimed skill to bosses. Keep at -1 if Config.AttackSkill[1] is untimed skill.
	Config.AttackSkill[3] = 64; // Primary skill to others.
	Config.AttackSkill[4] = -1; // Primary untimed skill to others. Keep at -1 if Config.AttackSkill[3] is untimed skill.
	Config.AttackSkill[5] = -1; // Secondary skill if monster is immune to primary.
	Config.AttackSkill[6] = 47; // Secondary untimed skill if monster is immune to primary untimed.

	// Low mana skills - these will be used if main skills can't be cast.
	Config.LowManaSkill[0] = -1; // Timed low mana skill.
	Config.LowManaSkill[1] = -1; // Untimed low mana skill.
